server/internal/background/index_toolset.go

```diff
@@ -0,0 +1,67 @@
+package background
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"go.temporal.io/api/enums/v1"
+	"go.temporal.io/sdk/client"
+	"go.temporal.io/sdk/temporal"
+	"go.temporal.io/sdk/workflow"
+
+	"github.com/google/uuid"
+
+	"github.com/speakeasy-api/gram/server/gen/types"
+	"github.com/speakeasy-api/gram/server/internal/background/activities"
+)
+
+type IndexToolsetParams struct {
+	ProjectID   uuid.UUID
+	ToolsetSlug types.Slug
+}
+
+type IndexToolsetClient struct {
+	Temporal client.Client
+}
+
+func ExecuteIndexToolset(
+	ctx context.Context,
+	temporalClient client.Client,
+	params IndexToolsetParams,
+) (client.WorkflowRun, error) {
+	id := fmt.Sprintf(
+		"v1:index-toolset:%s:%s",
+		params.ProjectID,
+		params.ToolsetSlug,
+	)
+	return temporalClient.ExecuteWorkflow(ctx, client.StartWorkflowOptions{
+		ID:                    id,
+		TaskQueue:             string(TaskQueueMain),
```

Comment:
I would add a conflict policy here. We don't need multiple of these running a time for the same toolset. Set the conflict policy so it only allows one running, the existing one